runtime/doc/treesitter.txt

```diff
@@ -947,6 +947,31 @@ set({lang}, {query_name}, {text})                 *vim.treesitter.query.set()*
       • {text}        (string) Query text (unparsed).
 
 
+==============================================================================
+Lua module: vim.treesitter.query_linter          *lua-treesitter-query_linter*
+
+clear({buf})                             *vim.treesitter.query_linter.clear()*
+    Clears diagnostics of the query linter for a given buffer
+
+    Parameters: ~
+      • {buf}  integer Buffer to clear diagnostics
+
+lint({buf}, {opts})                       *vim.treesitter.query_linter.lint()*
+    Determines diagnostics for a buffer containing tree-sitter queries
+    (|treesitter-query|). For runtime query files that target installed
+    tree-sitter parsers, this function will use tree-sitter API to determine
+    admissible identifiers to describe nodes of those parsers and verifies
+    that top-level s-expressions can be loaded in isolation by Neovim to debug
+    errors in runtime query files. The found diagnostics are reported using
+    |diagnostic-api|.
+
+    Parameters: ~
+      • {buf}   integer Buffer to lint
+      • {opts}  nil|table Options for linting
+                • langs nil|string|string[]|function One or more parsers which
```

Comment:
`string|string[]|function` overly complicated? Is `string|strings[]` enough? For the ftplugin, the user could define the target language with a buffer variable.

Reply:
Do you have a concrete use case for dynamic language? Otherwise I'd propose leaving this out for now -- widening the signature is always possible without breaking, while the converse is not.

Reply:
removed functions from official documentation.